app/domain/authentication/authn_iam/authenticator.rb

```diff
@@ -0,0 +1,95 @@
+require 'net/http'
+require 'net/https'
+require 'json'
+
+module Authentication
+  module AuthnIam
+    class Authenticator
+      
+      def initialize(env:)
+        @env = env
+      end
+
+      def valid?(input)
+        @account, @login, password = input.account, input.username, input.password
+
+        # JSON holding the AWS signed headers 
+        signed_aws_headers = JSON.parse password
+
+        response_hash = identity_response_hash(signed_aws_headers)
+        trusted = response_hash != false
+
+        (trusted) && (iam_role_matches? host_role, response_hash)
+
+      end
+
+      def identity_response_hash(signed_aws_headers)
+
+        res = retrieve_iam_identity(signed_aws_headers)
+
+        Rails.logger.info("****> #{res.code} #{res.message}")
+        Rails.logger.info("**** Body -> #{res.body} ")
+
+        if res.code.eql?("200")
+          Hash.from_xml(res.body)
+        else
+          Rails.logger.error("****> #{res.code} #{res.message}")
+          false
+        end      
+
+      end
+    
+      def iam_role_matches? resource, response_hash
+    
+        return false if resource.nil?
+
+        is_allowed_role = false
+    
+        split_assumed_role = response_hash["GetCallerIdentityResponse"]["GetCallerIdentityResult"]["Arn"].split(":")
+
+        # removes the last 2 parts of login to be substituted by the info from getCallerIdentity
+        host_prefix = (@login.split("/")[0..-3]).join("/")
+        aws_account_id = split_assumed_role[4]
+        aws_role_name = split_assumed_role[5].split("/")[1]
+
+        host_to_match = "#{host_prefix}/#{aws_account_id}/#{aws_role_name}"
+
+        Rails.logger.info("host to match = #{host_to_match}")
```

Comment:
This a `debug` level message. Debug info is also elsewhere in this PR as `info`, so I'd like to see a cleanup of the logging.

Reply:
Cleaned up and messages still present were sent to debug